setup-env

```diff
@@ -9,61 +9,121 @@ USAGE=$(
 Configure a development environment for this repository.
 
 It does the following:
+  - Allows the user to specify the Python version to use for the virtual environment.
+  - Allows the user to specify a name for the virtual environment.
   - Verifies pyenv and pyenv-virtualenv are installed.
-  - Creates a Python virtual environment.
+  - Creates the Python virtual environment.
   - Configures the activation of the virtual enviroment for the repo directory.
   - Installs the requirements needed for development.
   - Installs git pre-commit hooks.
   - Configures git upstream remote "lineage" repositories.
 
 Usage:
-  setup-env [options] [virt_env_name]
+  setup-env (-n | --name) [virtual_env_name] (-v | --version) [python_version]
   setup-env (-h | --help)
 
 Options:
-  -f --force          Delete virtual enviroment if it already exists.
-  -h --help           Show this message.
-  -i --install-hooks  Install hook environments for all environments in the
-                      pre-commit config file.
+  -f | --force             Delete virtual enviroment if it already exists.
+  -h | --help              Show this message.
+  -i | --install-hooks     Install hook environments for all environments in the
+                           pre-commit config file.
+  -l | --list-versions     List available Python versions and select interactively.
+  -n | --name              Specify the name of the virtual environment.
+  -v | --version           Specify the Python version for the virtual environment.
 
 END_OF_LINE
 )
 
+# Display installed python versions
+python_versions() {
+  pyenv versions --bare --skip-aliases --skip-envs
+}
+
 # Flag to force deletion and creation of virtual environment
 FORCE=0
 
-# Positional parameters
-PARAMS=""
+# Initialize the other flags
+INSTALL_HOOKS=0
+LIST_VERSIONS=0
+PYTHON_VERSION=""
+VENV_NAME=""
+
+# Define long options
+LONGOPTS="force,help,install-hooks,list-versions,name:,version:"
+
+# Define short options for getopt
+SHORTOPTS="fhiln:v:"
+
+# Check if GNU getopt is available
+if [[ $(getopt --version 2> /dev/null) != *"getopt from util-linux"* ]]; then
```

Comment:
@mcdonnnj I addressed this here [0be1f63](https://github.com/cisagov/skeleton-generic/pull/166/commits/0be1f6376dc914be4eff9ce5fde7649d085a2582)

Reply:
I like the explanation added in https://github.com/cisagov/skeleton-generic/pull/166/commits/0be1f6376dc914be4eff9ce5fde7649d085a2582 but I would like a little bit more on the comment before the `if` statement. Checking the output of `getopt --version` against a regex is awkward and I would like a bit of documentation around it because, although unlikely, a change in the string's format would potentially break this check.

Reply:
Oh yeah sorry I see what you mean. I will git it fixed

Reply:
@mcdonnnj see here: https://github.com/cisagov/skeleton-generic/pull/166/commits/f8824c8aa157cbee0ec95367b1754eeac67934c9

Reply:
That works for me.